.circleci/config.yml

```diff
@@ -4,31 +4,36 @@ jobs:
     docker:
       - image: linuxbrew/linuxbrew
     environment:
-      CIRCLE_REPOSITORY_URL: https://github.com/linuxbrew/homebrew-xorg
       HOMEBREW_DEVELOPER: 1
       HOMEBREW_NO_AUTO_UPDATE: 1
       HOMEBREW_VERBOSE: 1
       HOMEBREW_VERBOSE_USING_DOTS: 1
     steps:
-      - run: |
-          cd /home/linuxbrew/.linuxbrew/Homebrew
-          git fetch origin --unshallow --tags
-          git reset --hard origin/master
-      - checkout
-      - run: git remote set-url origin $CIRCLE_REPOSITORY_URL
-      - run: if [ -e .git/shallow ]; then echo git fetch --unshallow; fi
-      - run: git fetch origin
-      - run: git config --global user.name LinuxbrewTestBot
-      - run: git config --global user.email testbot@linuxbrew.sh
-      - run: mkdir -p /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/linuxbrew
-      - run: cp -a . /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/linuxbrew/$CIRCLE_PROJECT_REPONAME
-      - run: |
-          mkdir /tmp/bottles
-          cd /tmp/bottles
-          umask 022
-          PATH="/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH"
-          brew install patchelf
-          brew test-bot --tap=linuxbrew/xorg --bintray-org=linuxbrew --git-name=LinuxbrewTestBot --git-email=testbot@linuxbrew.sh
+      - run:
+          name: Set up Git
+          command: |
+            git config --global user.name LinuxbrewTestBot
+            git config --global user.email testbot@linuxbrew.sh
+      - run:
+          name: Set up linuxbrew/xorg tap
+          working_directory: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/linuxbrew
+          command: |
+            git clone https://github.com/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME
+            cd $CIRCLE_PROJECT_REPONAME
+            if [[ -n "$CIRCLE_PR_NUMBER" ]]; then
+                git fetch origin pull/$CIRCLE_PR_NUMBER/head
+            fi
+            git reset --hard $CIRCLE_SHA1
+      - run:
```

Comment:
How about:
```bash
            if [[ -n "$CIRCLE_PR_NUMBER" ]]; then
                git pull origin pull/$CIRCLE_PR_NUMBER/head
            fi
```
?

Reply:
We can also use
```
$CIRCLE_BRANCH/head
``` 
instead of
```
pull/$CIRCLE_PR_NUMBER/head
```

Reply:
`pull/$CIRCLE_PR_NUMBER/head` is more clear to me. `$CIRCLE_BRANCH/head` doesn't work when `$CIRCLE_BRANCH` is `master`. Neither does `pull/$CIRCLE_PR_NUMBER/head`, but it's more clear why it doesn't work.

Reply:
`git pull` creates a merge commit, which we definitely don't want.

Reply:
default is `--ff-only`, so it would be fine

Reply:
no, default is `-ff`, but in this case it should work anyway :)

Reply:
One option may be…
```
git fetch origin $CIRCLE_SHA1:$CIRCLE_BRANCH
git checkout $CIRCLE_BRANCH
```
I think I still prefer `git reset --hard $CIRCLE_SHA1` over `git checkout`

Reply:
The default behaviour of `git pull` is to create a merge commit, which we don't want.

Reply:
```
$ git help pull
...
       --ff
           When the merge resolves as a fast-forward, only update the branch pointer, without creating a merge commit. This is the default behavior.
```

Reply:
`When the merge resolves as a fast-forward`, when it doesn't, it creates a merge commit, which we don't want.

Reply:
I'm fine with resetting as it brings us to the same place...
but `git reset --hard` usually signifies that something has gone wrong and we're trying to fix it.

I thought PRs always resolve in fast-forwards... don't they? :confused: If they don' - sure, let's keep `reset --hard`. But if they are supposed to resolve in fast-forwards,then  we can do `--ff-only` which will fail if a merge commit is necessary and, therefore, it will serve as an additional checkpoint.

Reply:
If the PR is say one commit behind `origin/master`, then it won't resolve in a fast-forward. We can ask the submitter to rebase onto master, but it's nice not to have to if it's not needed.

Reply:
Ah! Gotcha! OK, I'm leaving this part untouched.